core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -2110,14 +2110,14 @@ private CloseableIterator<SchemaRegistryValue> allVersionsFromAllContexts(
   @Override
   public void close() throws IOException {
     log.info("Shutting down schema registry");
+    if (leaderRestService != null) {
+      leaderRestService.close();
+    }
```

Comment:
are we closing the leader elector last now?

Reply:
looks like you moved the code to close `leaderRestService` first? looks like the `leaderElector` itself is closed last? shouldn't we be closing the `leaderElector` first?

Reply:
Thanks for catching that, updated

Reply:
i think we need to close the `leaderElector` first followed by `leaderRestService`. this will avoid the split-brain problem. otherwise, you can have two pods think they're the leader and handle writes instead of forwarding the request to the right leader. by shutting down the `leaderRestService`, you'll potentially return 5xx for write requests which the client can retry instead of incorrectly handling the request. thoughts?

Reply:
`leaderRestService` is used by the follower to talk to leader, if we close `leaderElector`, the current pod will leave the consumer group so it won't be the leader anymore. 
But I think it doesn't harm to close `leaderRestService` after `leaderElector`.

Reply:
you're right. thanks!